exp/controllers/awsmachinepool_controller.go

```diff
@@ -415,6 +448,164 @@ func (r *AWSMachinePoolReconciler) reconcileDelete(machinePoolScope *scope.Machi
 	return nil
 }
 
+func reconcileDeleteAWSMachines(ctx context.Context, mp *expclusterv1.MachinePool, clusterScope cloud.ClusterScoper, client client.Client) error {
+	awsMachineList, err := getAWSMachines(ctx, mp, client)
+	if err != nil {
+		return err
+	}
+	for i := range awsMachineList.Items {
+		awsMachine := awsMachineList.Items[i]
+		if !awsMachine.DeletionTimestamp.IsZero() {
+			// delete the owner Machine resource for the AWSMachine so that CAPI can clean up gracefully
+			machine, err := util.GetOwnerMachine(ctx, client, awsMachine.ObjectMeta)
+			if err != nil {
+				clusterScope.Warn("failed to get owner machine", "awsmachine", awsMachine, "machinepool", mp.Name, "namespace", mp.Namespace)
```

Comment:
It could have no owner for legitimate reasons, e.g if the awsmachine was just provisioned by an ASG, CAPA reconciled and saw it, created an AWSMachine. Meanwhile, CAPI hasn't reconciled and hasn't yet created a machine for the new instance.

I _suppose_ we could distinguish between the two. As if it's the above scenario, the owner reference will be the AWSMachinePool. Then we could error and leave it in place if it was in some bizarre unknown state. Do you think it's worth the effort?